doc/api/process.md

```diff
@@ -2,97 +2,131 @@
-exiting of the event loop at this point, and once all `'exit'` listeners have
-finished running the process will exit. Therefore you **must** only perform
-**synchronous** operations in this handler. This is a good hook to perform
-checks on the module's state (like for unit tests). The callback takes one
-argument, the code the process is exiting with.
+The `'exit'` event is emitted when the Node.js process is about to exit as a
+result of either:
+
+* The `process.exit()` method being called explicitly;
+* The Node.js event loop no longer having any additional work to perform.
+
+There is no way to prevent the exiting of the event loop at this point, and once
+all `'exit'` listeners have finished running the Node.js process will terminate.
 
-This event is only emitted when Node.js exits explicitly by process.exit() or
-implicitly by the event loop draining.
+The listener callback function is invoked with the exit code specified either
+by the [`process.exitCode`][] property, or the `exitCode` argument passed to the
+[`process.exit()`] method, as the only argument.
+
+For example:
+
+```js
+process.on('exit', (code) => {
+  console.log(`About to exit with code: ${code}`);
+});
+```
 
-Example of listening for `'exit'`:
+Listener functions **must** only perform **synchronous** operations. The Node.js
+process will exit immediately after calling the `'exit'` event listeners
+causing any additional work still queued in the event loop to be abandoned.
+In the following example, for instance, the timeout will never occur:
 
 ```js
 process.on('exit', (code) => {
-  // do *NOT* do this
   setTimeout(() => {
     console.log('This will not run');
   }, 0);
-  console.log('About to exit with code:', code);
 });
 ```
 
-## Event: 'message'
+### Event: 'message'
 <!-- YAML
 added: v0.5.10
 -->
 
+If the Node.js process is spawned with an IPC channel (see the [Child Process][]
+and [Cluster][] documentation), the `'message'` event is emitted whenever a
+message sent by a parent process using [`childprocess.send()`][] is received by
+the child process.
+
+The listener callback is invoked with the following arguments:
 * `message` {Object} a parsed JSON object or primitive value
 * `sendHandle` {Handle object} a [`net.Socket`][] or [`net.Server`][] object, or
   undefined.
 
-Messages sent by [`ChildProcess.send()`][] are obtained using the `'message'`
-event on the child's process object.
 
-## Event: 'rejectionHandled'
+### Event: 'rejectionHandled'
 <!-- YAML
 added: v1.4.1
 -->
 
-Emitted whenever a Promise was rejected and an error handler was attached to it
-(for example with [`promise.catch()`][]) later than after an event loop turn. This event
-is emitted with the following arguments:
+The `'rejectionHandled'` event is emitted whenever a `Promise` has been rejected
+and an error handler was attached to it (using [`promise.catch()`][], for
+example) later than one turn of the Node.js event loop.
```

Comment:
"later than one turn of the event loop" isn't correct - but I'm not sure we want to get into the real specifics, probably better to lie here.